### ISSUE TITLE
Suggestion: cursor TTL should be relative to last read

### ISSUE THREAD

**Issue by miguel-porto** [arangodb/arangodb#4490]
Hi, this is just a suggestion, I think the TTL of a cursor should be relative to last read, not to the time at which it was created. I have a long running query (cause I have heavy processing upon each read that takes about 1 second between reads), and it breaks in the middle due to cursor timeout. Of course I can increase the TTL, but the point is, I think it would make more sense that TTL would be relative to last read. What do you think?
Best.
Miguel

**Comment by jsteemann** [comment]
@KevinLvLight : yes, cursor timeouts are renewed upon each request to the cursor. This is the case since several versions of ArangoDB already.
If you still think there is a problem, can you provide an exact use scenario that shows the problem?
Thanks!

**Comment by KevinLvLight** [comment]
Thanks!
I'll try to upgrade the version to see if the problem can be solved
P.S. Could you please let me know which version to start with？

**Comment by MaorDavidzon** [comment]
@jsteemann Could you please provide information on the version in which the issue was resolved?
It happens to us in version 3.11.6 in Arango cloud.
